Mathlib/Topology/Algebra/Module/WeakDual.lean

```diff
@@ -314,6 +314,45 @@ instance instTopologicalSpace : TopologicalSpace (WeakSpace 𝕜 E) :=
 instance instContinuousAdd : ContinuousAdd (WeakSpace 𝕜 E) :=
   WeakBilin.instContinuousAdd (topDualPairing 𝕜 E).flip
 
+
+/-- There is a canonical map `E → WeakSpace 𝕜 E` (the "identity"
+mapping). It is a linear equivalence. -/
+def toWeakSpace : E ≃ₗ[𝕜] WeakSpace 𝕜 E := LinearEquiv.refl 𝕜 E
+-- align?
+
+@[simp]
+-- theorem coe_toWeakSpace (x : E) : (toWeakSpace x : WeakSpace 𝕜 E) = x :=
+theorem coe_toWeakSpace (x : E) : toWeakSpace x = x :=
+  rfl
+-- #align ?
+
+@[simp]
+theorem toWeakSpace_eq_iff (x y : E) : (toWeakSpace x : WeakSpace 𝕜 E) = toWeakSpace y ↔ x = y :=
+  Function.Injective.eq_iff <| LinearEquiv.injective toWeakSpace
+-- #align ?
+
+-- theorem toWeakSpace_continuous : Continuous fun x : E => (toWeakSpace x : WeakSpace 𝕜 E) := by
+theorem toWeakSpace_continuous : Continuous fun x : E => toWeakSpace x := by
+  refine { isOpen_preimage := ?isOpen_preimage }
+  intro U hU
+  apply Continuous.le_induced _ U hU
+  apply continuous_pi_iff.mpr
+  intro y
+  exact ContinuousLinearMap.continuous y
+
+
+/-- For a topological vector space `E`, according to `toWeakSpace_continuous` the "identity mapping"
+`E → WeakSpace 𝕜 E` is continuous. This definition implements it as a continuous linear map. -/
+def continuousLinearMapToWeakSpace : E →L[𝕜] WeakDual 𝕜 E :=
+  { toWeakSpace with cont := toWeakSpace_continuous }
+-- #align ?
```

Comment:
I would prove the continuity inside the definition and use the proof
```
    apply WeakBilin.continuous_of_continuous_eval
    exact ContinuousLinearMap.continuous
```

Reply:
Thanks for the code, but I cannot go into the tactic mode even though I write
```lean
def continuousLinearMapToWeakSpace : E →L[𝕜] WeakDual 𝕜 E :=
  { toWeakSpace with cont := by
    apply WeakBilin.continuous_of_continuous_eval -- expected '{' or indented tactic sequence
    exact ContinuousLinearMap.continuous }
```

Instead, if I write separately the continuity as above, I get the error
```lean
def continuousLinearMapToWeakSpace : E →L[𝕜] WeakDual 𝕜 E :=
  { toWeakSpace with cont := toWeakSpace_continuous }
 -- type mismatch
  toWeakSpace_continuous
has type
  Continuous fun x ↦ toWeakSpace x : Prop
but is expected to have type
  Continuous ?m.145022.toFun : Prop
```

Your help is much appreciated!

Reply:
You need two more spaces in the first code-block. However the nicer way to write this is using `where` as written in https://leanprover-community.github.io/contribute/style.html#structuring-definitions-and-theorems (last paragraph of that section)

Reply:
thank you, I could use the tactic mode. I wasn't sure about where, whether I would have to fill in all other fields or there was other ways. Is there any way to add only `cont` by extending `LinearMap`?

Reply:
You can write `__ := toWeakSpace`

Reply:
ok, done!